### ISSUE TITLE
[jsconsole] personal.sign(...) fails even though personal_sign API is available and working

### ISSUE THREAD

**Issue by chris-j-h** [ConsenSys/quorum#1190]
**To Reproduce:**

1. From the attached js console of node 1 of the 7nodes example, Quorum **v21.4.0+**

```
> personal.sign("0xaaaaaa","0xed9d02e382b34818e88b88a309c7fe71e65f419d","")

GoError: Error: invalid address at web3.js:3930:11(47)
	at native
	at <eval>:1:71(5)
```

2. `personal_sign` RPC API works as expected

```
% curl -X POST -H "Content-Type:application/json" -d '{"id":1,"method":"personal_sign","params":["0xaaaaaa","0xed9d02e382b34818e88b88a309c7fe71e65f419d",""]}' http://localhost:22000

{"jsonrpc":"2.0","id":1,"result":"0xa661ee8721999743d40bcbefec309c262c10dcf159f60c3d1dfbb67b850007f010d926403753c6d949e132be4327f36344ffe4cade07c55fe229b1460c15e0481c"}
```

**Possible cause**

The v21.4.0 upgrade introduced commit [7b68975a](https://github.com/ConsenSys/quorum/commit/7b68975a00ee914d91f9c6a56c55a357012ab844) which changed the js console interpreter from otto to goja.

Reverting this commit makes `personal.sign` work as expected again.  A dev branch with this commit reverted can be found [here](https://github.com/chris-j-h/quorum/tree/21.4.0-jsconsole-reverted). 

The error is thrown by the [inputAddressFormatter](https://github.com/ConsenSys/quorum/blob/master/internal/jsre/deps/web3.js#L3930) applied to the address param.

Curiously none of the other APIs that use this formatter have the same issue (e.g. `eth.getCode`, `personal.unlockAccount`, ...).

**Comment by prd-fox** [comment]
As you mentioned, it was a bug in the JS upgrade. Looks to be fixed in geth 1.9.21.
https://github.com/ethereum/go-ethereum/issues/21417
https://github.com/ethereum/go-ethereum/pull/21503

**Comment by ricardolyn** [comment]
thanks for the info @chris-j-h @prd-fox . I will link this to a geth upgrade issue so we can get it soon fixed from upstream

**Comment by chris-j-h** [comment]
Fixed in 21.7.0 - https://github.com/ConsenSys/quorum/commit/ff23e265cd2ff13f33e145071b317fc25f0a2b23
